java_cucumber/src/test/java/java_cucumber/Stepdefs.java

```diff
@@ -930,4 +943,290 @@ public void newAccInfo() throws ApiException, NoSuchAlgorithmException, com.algo
+    }
+
+    @Then("the creator should have {int} assets remaining")
+    public void the_creator_should_have_assets_remaining(Integer expectedBal) throws ApiException {
+        com.algorand.algosdk.algod.client.model.Account accountResp = 
+                this.acl.accountInformation(this.creator);
+        AssetHolding holding = accountResp.getHolding(this.assetID);
+        Assert.assertEquals(BigInteger.valueOf(expectedBal), holding.getAmount());
+    }
+
+    @Then("I update the asset index")
+    public void i_update_the_asset_index() throws ApiException {
+        com.algorand.algosdk.algod.client.model.Account accountResp = acl.accountInformation(this.creator);
+        Set<java.math.BigInteger> keys = accountResp.getThisassettotal().keySet();
+        this.assetID = Collections.max(keys); 
+    }
+
+    @When("I send the bogus kmd-signed transaction")
+    public void i_send_the_bogus_kmd_signed_transaction() {
+        try {
+            txid = acl.rawTransaction(this.stxBytes).getTxId();
+        } catch (ApiException e) {
+            this.err = true;
+        }
+    }
+
+    @Then("I create a transaction for a second account, signalling asset acceptance")
+    public void i_create_a_transaction_for_a_second_account_signalling_asset_acceptance() throws ApiException, NoSuchAlgorithmException {
+        
+        TransactionParams params = acl.transactionParams();
+        lastRound = params.getLastRound();
+
+        Transaction tx = Transaction.createAssetAcceptTransaction(
+                new Address(this.accounts.get(1)),
+                acl.transactionParams().getFee(), // transaction fee
+                acl.transactionParams().getLastRound(), // first valid round 
+                acl.transactionParams().getLastRound().add(BigInteger.valueOf(1000)), // last valid round
+                this.note,
+                acl.transactionParams().getGenesisID(),
+                new Digest(acl.transactionParams().getGenesishashb64()),
+                this.assetID);
+        Account.setFeeByFeePerByte(tx, tx.fee);
+        this.txn = tx;
+        this.pk = new Address(this.accounts.get(1));
+    }
+
+    @Then("I send the kmd-signed transaction")
+    public void i_send_the_kmd_signed_transaction() throws ApiException {
+        txid = acl.rawTransaction(this.stxBytes).getTxId();
+    }
+
+    @When("I create a freeze transaction targeting the second account")
+    public void i_create_a_freeze_transaction_targeting_the_second_account() throws NoSuchAlgorithmException, ApiException, com.algorand.algosdk.kmd.client.ApiException {
+        this.renewHandle(); // to avoid handle expired error
+        TransactionParams params = acl.transactionParams();
+        lastRound = params.getLastRound();
+
+        Transaction tx = Transaction.createAssetFreezeTransaction(
+                new Address(this.accounts.get(0)), // transaction sender
+                new Address(this.accounts.get(1)), // revoked from
```

Comment:
I don't think this transaction is revoking any assets. Is this the account being frozen?

Reply:
the comment should say "account to freeze" instead of "revoked from"